src/System.Runtime.Analyzers/Core/AvoidZeroLengthArrayAllocations.cs

```diff
@@ -83,6 +90,28 @@ private void AnalyzeOperation(OperationAnalysisContext context)
             }
         }
 
+        private static bool IsCompilerGeneratedParamsArray(IArrayCreationExpression arrayCreationExpression, OperationAnalysisContext context)
+        {
+            var model = context.Compilation.GetSemanticModel(arrayCreationExpression.Syntax.SyntaxTree);
+
+            // Compiler generated array creation seems to just use the syntax from the parent.
+            var parent = model.GetOperation(arrayCreationExpression.Syntax, context.CancellationToken) as IInvocationExpression;
```

Comment:
@JohnHamby @genlu - we really need operation.GetParent(). Otherwise people will just do this which is very inefficient because this semantic model wouldn't have any bound trees cached on it and would redo the binding.